pkg/pipeline/lifecycle.go

```diff
@@ -93,6 +94,20 @@ func (s *Service) Start(
 
 	s.logger.Debug(ctx).Str(log.PipelineIDField, pl.ID).Msg("starting pipeline")
 
+	err = s.startInternal(ctx, pl, connFetcher, procFetcher, pluginFetcher)
+	s.notify(pl.ID, pl.ProvisionedBy, err)
+	return err
```

Comment:
Connected to the discussion about stopping Conduit if a pipeline fails to be created, I'll argue we shouldn't stop if the pipeline is not _started_* successfully. This code will cause Conduit to stop if I create a pipeline without any connectors and attempt to start it. Since I'd get immediate feedback about this through the API response, I don't think this should crash Conduit.

*Note that "started" in this context only means that the pipeline nodes are successfully created, connected together with channels, and all goroutines are started. If one of those goroutines stops _right after_ the start because of an error (e.g. can't establish the connection to the database), that's when Conduit should stop (that's done down in line 641).